mesonbuild/dependencies/pkgconfig.py

```diff
@@ -73,6 +83,158 @@ def list_all(self) -> T.List[str]:
         '''Return all available pkg-config modules'''
         raise NotImplementedError
 
+    @staticmethod
+    def get_env(environment: Environment, for_machine: MachineChoice,
+                uninstalled: bool = False) -> EnvironmentVariables:
+        env = EnvironmentVariables()
+        key = OptionKey('pkg_config_path', machine=for_machine)
+        extra_paths: T.List[str] = environment.coredata.options[key].value[:]
+        if uninstalled:
+            uninstalled_path = Path(environment.get_build_dir(), 'meson-uninstalled').as_posix()
+            if uninstalled_path not in extra_paths:
+                extra_paths.append(uninstalled_path)
+        env.set('PKG_CONFIG_PATH', extra_paths)
+        sysroot = environment.properties[for_machine].get_sys_root()
+        if sysroot:
+            env.set('PKG_CONFIG_SYSROOT_DIR', [sysroot])
+        pkg_config_libdir_prop = environment.properties[for_machine].get_pkg_config_libdir()
+        if pkg_config_libdir_prop:
+            env.set('PKG_CONFIG_LIBDIR', pkg_config_libdir_prop)
+        return env
+
+
+class PyPkgConf(PkgConfigInterface):
+    '''libpkgconf implementation'''
+
+    def __init__(self, env: Environment, for_machine: MachineChoice, silent: bool):
+        self.env = env
+        self.for_machine = for_machine
+        self.pypkgconf = None
+        try:
+            self._load_pypkgconf()
+        except ImportError as e:
+            mlog.debug(str(e))
+
+    def _load_pypkgconf(self) -> None:
+        import pypkgconf
+
+        mlog.log(f'Found pypkgconf: {mlog.bold(pypkgconf.pypkgconf_version())} (libpkgconf {pypkgconf.pkgconf_version()})', once=True)
+        self.pypkgconf = pypkgconf
+
+    @functools.lru_cache(maxsize=None)
+    def query(self,
+              name: T.Optional[str] = None,
+              want_flags: T.Optional[int] = None,
+              want_variable: T.Optional[str] = None,
+              define_variable: T.Optional[T.Tuple[str, str]] = None) -> T.List[str]:
+        if self.pypkgconf is None:
+            raise MesonException('pypkgconf is not configured')
```

Comment:
I think you can remove all those checks. We have the found() method already that guarantees we won't use it if pypkgconf is None. Or turn them into asserts if you wish.

Reply:
This could be an `assert`. But I need something, otherwise mypy think that `self.pypkgconf` can be `None`